csv-to-genesis/src/main/kotlin/jp/co/soramitsu/Converter.kt

```diff
@@ -0,0 +1,153 @@
+import jp.co.soramitsu.iroha2.generated.datamodel.Name
+import jp.co.soramitsu.iroha2.generated.datamodel.account.AccountId
+import jp.co.soramitsu.iroha2.generated.datamodel.asset.AssetDefinitionId
+import jp.co.soramitsu.iroha2.generated.datamodel.asset.AssetId
+import jp.co.soramitsu.iroha2.generated.datamodel.asset.AssetValueType
+import jp.co.soramitsu.iroha2.generated.datamodel.isi.Instruction
+import jp.co.soramitsu.iroha2.transaction.Instructions
+import jp.co.soramitsu.iroha2.transaction.TransactionBuilder
+import org.apache.commons.csv.CSVFormat
+import org.apache.commons.csv.CSVParser
+import org.apache.commons.csv.CSVRecord
+import java.io.File
+import java.math.BigInteger
+import java.net.URL
+import java.security.KeyPair
+import java.text.SimpleDateFormat
+import java.time.Duration
+import java.util.*
+
+class Converter {
+    companion object {
+        private val ID = "id".asName() to 0
+        private val FRAUD_TYPE = "ft".asName() to 12
+        private val ORIGINATION = "org".asName() to 4
+        private val DESTINATION = "dst".asName() to 5
+        private val STATUS = "sts".asName() to 14
+        private val TIMESTAMP = "ts".asName() to 1
+        private val EXPIRY_DATE = "ed".asName() to TIMESTAMP.second
+
+        val dateFormatter = SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
+    }
+
+    suspend fun sendToIroha(csv: File, peerUrl: URL, admin: AccountId, keyPair: KeyPair) {
+        val client = Iroha2Client(peerUrl, true)
+        csv.bufferedReader().use { reader ->
+            val parser = CSVParser(
+                reader, CSVFormat.DEFAULT
+                    .withFirstRecordAsHeader()
+                    .withSkipHeaderRecord()
+            )
+            parser.forEach {
+                client.send(
+                    *it.mapToAssetIsi(admin).toTypedArray(),
+                    account = admin,
+                    keyPair = keyPair
+                )
+                println("#${it.get(0)} saved")
+            }
+        }
+    }
+
+    fun toGenesis(csv: File, genesis: File, admin: AccountId): Genesis {
+        val node = JSON_SERDE.readTree(genesis)
+        val block = JSON_SERDE.convertValue(node, RawGenesisBlock::class.java)
+        val txs = block.transactions.cast<MutableList<GenesisTransaction>>()
+
+        csv.bufferedReader().use { reader ->
+            val parser = CSVParser(
+                reader, CSVFormat.DEFAULT
+                    .withFirstRecordAsHeader()
+                    .withSkipHeaderRecord()
+            )
+            val isi = mutableListOf<Instruction>()
+            parser.forEach { isi.addAll(it.mapToAssetIsi(admin)) }
+            txs.add(GenesisTransaction(isi))
+        }
+
+        return Genesis(RawGenesisBlock(txs))
+    }
+
+    private fun CSVRecord.mapToAssetIsi(accountId: AccountId): ArrayList<Instruction> {
+        val isi = mutableListOf<Instruction>()
+
+        val id = this.get(ID.second)
```

Comment:
ID must be smth like +79095551212-+79095551215 so `range_start,range_end` should be used with prepending `+`: https://github.com/soramitsu/orillion-core-middleware/blob/567afe2fb5806d835614509e4cd68b0f18ff15b8/libs/contracts/src/main/kotlin/jp/co/soramitsu/orillion/contract/Regex.kt#L5